### PR TITLE
Fix most x86 build breaks

### PR DESCRIPTION
**Pull request by BruceForstall**
Fixed DecomposeInd break.
Fixed all decomp function comments.
Fixed lsra break.

Didn't fix break in LEGACY_BACKEND in liveness.cpp related to fgPerNodeLocalVarLiveness(): I'll let you do that one.

@pgavlin PTAL

### CODE REVIEW

**Comment by BruceForstall** [comment]
actually, I'm fixing the liveness.cpp issues also and will re-push

**Comment by BruceForstall** [comment]
@pgavlin PTAL

File: src/jit/lsra.h [hunk 1]
```diff
@@ -584,7 +584,7 @@ class LinearScan : public LinearScanInterface
     bool            isRegCandidate(LclVarDsc* varDsc);
 
     // Return the registers killed by the given tree node.
-    static regMaskTP getKillSetForNode(Compiler* compiler, GenTree* tree);
```
Comment (pgavlin) [inline src/jit/lsra.h:587]: This needs to remain static: it's normally used in some debug checks in codegenxarch that I `#if 0`'d out for the time being.

Comment (pgavlin) [inline src/jit/lsra.h:587]: Was this breaking the build?

Comment (BruceForstall) [inline src/jit/lsra.h:587]: yes. x86 references some non-static fields

Comment (pgavlin) [inline src/jit/lsra.h:587]: Huh, I thought I had fixed that. I'll take another look after this is in.

**Pull request merged**